### PR TITLE
Fix ip address verify failed

### PR DESCRIPTION
**Pull request by xutian**

### CODE REVIEW

**Comment by xutian** [comment]
ID: 1181956

**Comment by xiagao** [comment]
Acked-by:Xiaoling Gao xiagao@redhat.com

File: virttest/utils_env.py [hunk 1]
```diff
@@ -46,43 +44,56 @@ def wrapper(*args, **kwargs):
...
-
-            env["address_cache"][mac_address] = ip_address
-            env["address_cache"]["time_%s" % mac_address] = time.time()
-            del env["address_cache"]["last_seen_mac"]
+    """
+    Update mac <-> ip releation ship dict when we got a dhcpack packet.
+    """
+    address_cache = env["address_cache"]
+    # counter to record how may line searched
+    count = address_cache.get("line_count", 0) + 1
```
Comment (CongSmile) [inline virttest/utils_env.py]: Hi @xutian 

Why +1 for count here?

Comment (xutian) [inline virttest/utils_env.py]: It's a counter use to cont how many lines reading from tcpdump and that mean how many times this function called from env_utils.
@xutian

File: virttest/utils_net.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def __str__(self):
 class TAPCreationError(NetError):
 
     def __init__(self, ifname, details=None):
-        NetError.__init__(self, ifname, details)
+        super(TAPCreationError, self).__init__()
```
Comment (CongSmile) [inline virttest/utils_net.py]: Hi @xutian 

Could you check the following error?    

File "/home/staf-kvm-devel/autotest-devel/client/tests/virt/virttest/utils_net.py", line 962, in open_tap
      raise TAPCreationError(ifname, details)
    File "/home/staf-kvm-devel/autotest-devel/client/tests/virt/virttest/utils_net.py", line 67, in __init__
      super(MacvtapCreationError, self).__init__()
  TypeError: super(type, obj): obj must be an instance or subtype of type

Thanks.

Comment (xutian) [inline virttest/utils_net.py]: @CongSmile 

it looks that a python's bug  http://bugs.python.org/issue20503.

But maybe we can call super class 's __init__ function like NetError.__init__(self) to workaround this issue.

Thanks,
Xu

File: virttest/virt_vm.py [hunk 3]
```diff
@@ -645,19 +646,20 @@ def get_address(self, index=0):
...
+                    arping_cmd = "%s -f -c 5 " % arping_bin
+                    arping_cmd += "-I %s %s" % (nic.netdst, ip)
+                    utils.system(arping_cmd, ignore_status=True)
                 ip_map = utils_net.parse_arp()
                 arp_ip = ip_map.get(nic.mac.lower())
-                if arp_ip:
-                    self.address_cache[nic.mac.lower()] = arp_ip
-
             if not arp_ip:
                 raise VMIPAddressMissingError(nic.mac)
```
Comment (CongSmile) [inline virttest/virt_vm.py]: Hi @xutian 

Could we add more info if the IP verification failed? especially the failed reasons which can help analyze the case.
e.g. the result of ping, failed or successfully?

Thanks.

**Comment by CongSmile** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Comment by ypu** [comment]
LGTM

Acked-by:Yiqiao Pu <ypu@redhat.com>

**Comment by ypu** [comment]
Thanks @xutian and @CongSmile will merge it.

**Pull request merged**